File: shared/src/main/java/com/batu/shared/dto/request/TransactionRequestDto.java
package com.batu.shared.dto.request;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class TransactionRequestDto {

    private UUID transactionId;
    private UUID userId;
    private UUID accountId;
    private BigDecimal amount;
    private String isoCurrencyCode;
    private String transactionName;
    private String transactionType;
    private LocalDate date;
    private Boolean pending = Boolean.FALSE;
    private String paymentChannel;
    private String detailedCategoryCode;
    @JsonProperty("isActive")
    private boolean isActive = true;
    private long syncVersion;

    public TransactionRequestDto(UUID transactionId, UUID userId, UUID accountId, BigDecimal amount, String isoCurrencyCode,
            String transactionName, String transactionType, LocalDate date, Boolean pending, String paymentChannel,
            String detailedCategoryCode, boolean isActive) {
        this(transactionId, userId, accountId, amount, isoCurrencyCode, transactionName, transactionType, date, pending,
                paymentChannel, detailedCategoryCode, isActive, 0L);
    }
}

File: shared/src/main/kotlin/com/batu/shared/dto/TransactionRequestDto.kt
package com.batu.shared.dto

import java.time.LocalDate
import java.math.BigDecimal
import java.util.UUID
import com.batu.shared.dto.TransactionDetailedCategoryRequestDto

data class TransactionRequestDto(
    val userId: UUID,
    val accountId: UUID,
    val externalId: String,
    val amount: BigDecimal,
    val isoCurrencyCode: String,
    val transactionName: String,
    val transactionType: String,
    val date: LocalDate,
    val is_pending: Boolean? = false,
    val paymentChannel: String,
    val detailedCategoryCode: String,
    val isActive: Boolean = true
)